github/VisualStudio
# Mark branch for created PR.

# Description

When we create a pull request using the GitHub pane, mark the branch in the `.git/config` to associate the branch with the pull request, meaning that the status bar shows the PR for the current branch.

# Testing

- Create a new branch
- Make some changes and commit
- Create a new Pull Request using the link in the PR list to create a new PR
- Previously, the new PR would not be highlighted in the PR list, and would not show in the status bar
- With this PR it should be highlighted and show in the status bar

# Implementation

To do this, I had to make `TeamExplorerContext` aware of the PR mark in the git config, and manually update `IVSGitExt` to notify it of the change. The required making `TextExplorerContext` dependent on `IPullRequestService` in order to not duplicate the code that adds the property to the git config.
